Add e2e test for connectivity adapter (#7161)
kyma-project/kyma: tests/end-to-end/external-solution-integration/README.md

11. Sends an event to the Application Gateway. 
12. Verifies if the call from the lambda reached the test service.

> ## run test locally

Reviewer: ```suggestion
### Run test locally
```

>     dexSecretNamespace: "kyma-system"

Reviewer: Why those lines are here and what they represent?

11. Sends an event to the Application Gateway. 
12. Verifies if the call from the lambda reached the test service.

### Compass connectivity adapter test

This scenario uses Compass and Connectivity Adapter to register an Application and its APIs and Events.

> Connectivity Adapter is a component which translates Application Registry and Application Connector REST API to Compass GraphQL API. 

Reviewer 2: ```suggestion
Connectivity Adapter is a component which translates the Application Registry and Application Connector REST API to the Compass GraphQL API. 
```

### Steps

The test performs the following actions:
1. Registers an empty Application in Compass.
2. Creates the ApplicationMapping CR for the created Application in the `connectivity-adapter-e2e` Namespace.
3. Deploys a lambda function in the `connectivity-adapter-e2e` Namespace.
4. Starts a test service in the `connectivity-adapter-e2e` Namespace. The lambda function calls it when it receives an event.

> 6. Connects the Application through the Connectivity Adapter with the client certificates. 

Reviewer: Hmmm, should be 5 and then 6 ;)

> 6. Registers Service for the Application using Connectivity Adapter

Reviewer 2: ```suggestion
6. Registers a service for the Application using Connectivity Adapter.
```

7. Creates ServiceInstances for ServiceClasses registered by Compass Runtime Agent (one for the API and one for the Event services).
8. Creates a ServiceBinding for the API ServiceInstance.
9. Creates the ServiceBindingUsage CR of that binding for the deployed lambda function. 
10. Creates a Subscription for the lambda function, so that it is subscribed to the events exposed by the Application.
11. Sends an event to the Application Gateway. 
12. Verifies if the call from the lambda reached the test service.

> ## Environment variables for connectivity-adapter-e2e and compass e2e

Reviewer 2: ```suggestion
## Environment variables for connectivity-adapter-e2e and compass-e2e
```

## Run the test locally

### Run against Kyma cluster on Minikube 
1. Add an entry to your system's `/etc/hosts` that maps the `counter-service-{testID}.kyma.local` to `minikube cluster IP`,

> e.g.: for *compass-e2e-test*, it will be counter-service-compass-e2e-test.kyma.local 

Reviewer 2: ```suggestion
1. Add an entry to your system's `/etc/hosts` that maps the `counter-service-{testID}.kyma.local` to `minikube cluster IP`. For example, for `compass-e2e-test`, map `counter-service-compass-e2e-test.kyma.local`. 
```

2. Set required ENVs
3. Run the test using the following command:
    ```

>     go run ./cmd/runner {name-of-test}

Reviewer 2: ```suggestion
    go run ./cmd/runner {test-name}
```
